src/kzg.rs

```diff
@@ -0,0 +1,144 @@
+use crate::bytes::write_bytes_to_lower_hex;
+use crate::primitives::Bytes32;
+use crate::serde::{try_bytes_from_hex_str, HexError};
+use crate::ssz::ByteVector;
+use core::fmt;
+use ssz_rs::prelude::*;
+use std::ops::{Deref, DerefMut};
+use thiserror::Error;
+
+const KZG_COMMITMENT_BYTES_LEN: usize = 48;
+const KZG_PROOF_BYTES_LEN: usize = 48;
+
+pub type VersionedHash = Bytes32;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[cfg(feature = "serde")]
+    #[error("error deserializing hex-encoded input: {0}")]
+    Hex(#[from] HexError),
+    #[error(
+        "invalid length of encoding: expected {expected} bytes but only provided {provided} bytes"
+    )]
+    EncodingError { provided: usize, expected: usize },
+}
```

Comment:
do we need to keep this custom `Error` if we just use the simple type aliases below?

Reply:
Removed :) the less code the better